tensorrt_llm/_torch/auto_deploy/transformations/library/rope.py

```diff
@@ -0,0 +1,465 @@
+import operator
+from collections import defaultdict
+from typing import List
+
+import torch
+from torch.fx import GraphModule
+
+from ...utils.logger import ad_logger
+from ...utils.node_utils import bfs, identify_regions_between_residuals, is_op
+from .._graph import canonicalize_graph
+
+# Caches for fused cosine-sine tensors and position IDs.
+_rope_flash_cache = defaultdict(lambda: None)
```

Comment:
Let's not make it a global variable. I don't see any value in that, right? 

I don't think there is a world where v1 and v2 is mixed and even if it's mixed in some model implementations, I would wanna keep it separate as those are separate transformations... 

You can just initialize them in `match_rope_v1` and `match_rope_v2` and pass it as argument to the utility functions